### PR TITLE
spark: spark properties facet

### PR DESCRIPTION
**Pull request by tnazarew**
### Problem

There was a need to have a facet containing spark properties, with option for user to decide which properties should be included. 

Closes: #1673 

### Solution

Creating the mechanism to generate such facet.

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [x] You've updated any relevant documentation (_if relevant_)
- [x] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [x] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

File: integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/SparkPropertyFacet.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*
+/* Copyright 2018-2022 contributors to the OpenLineage project
```
Comment (pawel-big-lebowski) [inline integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/SparkPropertyFacet.java:2]: 2023 ;-)

File: integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/SparkPropertyFacetBuilder.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import org.apache.spark.SparkConf;
+import org.apache.spark.scheduler.SparkListenerJobStart;
+
+public class SparkPropertyFacetBuilder
+    extends CustomFacetBuilder<SparkListenerJobStart, SparkPropertyFacet> {
+  private static final Set<String> DEFAULT_ALLOWED_PROPERTIES =
+      new HashSet<>(Arrays.asList("spark.master", "spark.app.name"));
+  private static final String ALLOWED_PROPERTIES_KEY = "spark.openlineage.allowerProperties";
+  private final SparkConf conf;
+  private final Set<String> allowerProperties;
```
Comment (pawel-big-lebowski) [inline integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/SparkPropertyFacetBuilder.java:27]: alloweR

**Review by pawel-big-lebowski — COMMENTED** [review]
Please mind modifying CHANGELOG and documentation to let users know about that great feature.

File: integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/SparkPropertyFacet.java (hunk 1 above)
Comment (tnazarew) [reply integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/SparkPropertyFacet.java:2]: I'm going to leave this one be, we will need to change all of the headers at once so we can modify the verification.

File: integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/SparkPropertyFacetBuilder.java (hunk 2 above)
Comment (tnazarew) [reply integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/SparkPropertyFacetBuilder.java:27]: done

**Review by pawel-big-lebowski — APPROVED** [review]

**Pull request merged**